auth/user_mgt.go

```diff
@@ -107,83 +97,236 @@ type UserIterator struct {
 
 // UserToCreate is the parameter struct for the CreateUser function.
 type UserToCreate struct {
-	params map[string]interface{}
+	_req   *identitytoolkit.IdentitytoolkitRelyingpartySignupNewUserRequest
+	fields struct {
+		UID, DisplayName, Email, PhotoURL, PhoneNumber bool
+	}
 }
 
-func (u *UserToCreate) set(key string, value interface{}) {
-	if u.params == nil {
-		u.params = make(map[string]interface{})
+func (u *UserToCreate) request() *identitytoolkit.IdentitytoolkitRelyingpartySignupNewUserRequest {
+	if u._req == nil {
+		u._req = &identitytoolkit.IdentitytoolkitRelyingpartySignupNewUserRequest{}
 	}
-	u.params[key] = value
+	return u._req
+}
+
+func (u *UserToCreate) validatedRequest() (*identitytoolkit.IdentitytoolkitRelyingpartySignupNewUserRequest, error) {
+	req := u.request() // it is allowed to create a user without any parameters
```

Comment:
Tiny nit:
// creating a user without any parameters is allowed

Otherwise, it's ambiguous what "it" means.

Reply:
Done